src/psyclone/psyir/symbols/symboltable.py

```diff
@@ -180,40 +221,124 @@ def _normalize(key):
+            symbol name. This will be appended with an integer if the name \
+            clashes with an existing symbol name.
+        :type root_name: str or NoneType
+        :param str tag: optional tag identifier for the new symbol.
+        :param bool shadowing: optional logical flag indicating whether the \
+            name can be overlapping with a symbol in any of the ancestors \
+            symbol tables. Defaults to False.
+        :param symbol_type: class type of the new symbol.
+        :type symbol_type: Class type :py:class:`psyclone.psyir.symbols.Symbol`
+        :param symbol_init_args: arguments to create a new symbol.
+        :type symbol_init_args: unwrapped Dict[str] = object
+
+        '''
+
+        # Only type-check symbol_type, the other arguments are just passed down
+        # and type checked inside each relevant method.
+        if symbol_type:
+            if not (isinstance(symbol_type, type) and
+                    Symbol in inspect.getmro(symbol_type)):
+                raise TypeError(
+                    "The symbol_type parameter should be of type Symbol or "
+                    "one of its sub-classes but found '{0}' instead."
+                    .format(type(symbol_type)))
+        else:
+            symbol_type = Symbol
+
+        available_name = self.next_available_name(root_name, shadowing)
+        symbol = symbol_type(available_name, **symbol_init_args)
+        self.add(symbol, tag)
+        return symbol
+
+    def symbol_from_tag(self, tag, root_name=None, **new_symbol_args):
+        ''' Lookup a tag, if it doesn't exist create a new symbol with the
+        given tag. By default it creates a generic Symbol with the tag as the
+        root of the symbol name. Optionally, a different root_name or any of
+        the arguments available in the new_symbol() method can be given to
+        refine the name and the type of the created Symbol.
+
+        :param str tag: tag identifier.
+        :param str root_name: optional name of the new symbols if it needs \
+            to be created. Otherwise it is ignored.
+        :param new_symbol_args: arguments to create a new symbol.
+        :type new_symbol_args: unwrapped Dict[str, object]
+
+        :returns: symbol associated with the given tag.
+        :rtype: :py:class:`psyclone.psyir.symbols.Symbol`
+        '''
+        try:
+            symbol = self.lookup_with_tag(tag)
+            # Check that the symbol found matches the requested description
+            if 'symbol_type' in new_symbol_args:
+                symbol_type = new_symbol_args['symbol_type']
+                if not isinstance(symbol, new_symbol_args['symbol_type']):
+                    raise SymbolError(
+                        "Expected symbol with tag '{0}' to be of type '{1}' "
+                        "but found type '{2}'.".format(
+                        tag, symbol_type.__name__, type(symbol).__name__))
```

Comment:
pylint complains about the indentation of this line and 330.

Reply:
Done